File: interpreter.rb
require 'rubygems'
require 'treetop'
require 'parsimon_node_classes'
require 'pp'

Treetop.load "parsimon"

s1 = "
Node = record { left ^Node, right ^Node, key, value }
record Node { left ^Node, right ^Node, key, value }

main = 0
"
s2 = "
Node = record { left ^Node, right ^Node, key, value }
main = func(x,y,z) {
	5 + 11
}
"

parser = ParsimonParser.new
result = parser.parse(s2)
if result
	puts result.compile
else
	p parser.failure_reason
	p "#{parser.failure_line}, #{parser.failure_column}"
end

File: js/interpreter.js
var PEG = require('pegjs'),
	fs = require('fs');

fs.readFile("parsimon.pegjs", "UTF-8", function(err, data) {
	var parser = PEG.buildParser(data);

	var e2 = "record { left ^Node, right, key, value }";
	var r2 = parser.parse(e2);
	console.log(r2);

	var e3 = "func(x,y) { }";
	var r3 = parser.parse(e3);
	console.log(r3);

	var e4 = "Node = record { left ^Node, right ^Node, key, value }";
	var r4 = parser.parse(e4);
	console.dir(JSON.stringify(r4));
})